### PR TITLE
fix: forward record with proper history

### PR DESCRIPTION
**Pull request by Zangetsu101**
## Description

When indexing the validated record, the DECLARATION_UPDATED history entry wasn't being forwarded, which was causing conflicting assignments to be found during indexing and unassign.

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: packages/workflow/src/records/handler/validate.ts [hunk 1]
```diff
@@ -34,11 +35,12 @@ export const validateRoute = createRoute({
       request.payload
     )
 
-    const validatedRecord = await toValidated(
-      record,
-      token,
-      payload.comments,
-      payload.timeLoggedMS
+    await toValidated(record, token, payload.comments, payload.timeLoggedMS)
```
Comment (rikukissa) [inline packages/workflow/src/records/handler/validate.ts:38]: Is there an issue with `toValidated` if it doesn't return the same result an additional `getRecordById` would?

Comment (Zangetsu101) [reply packages/workflow/src/records/handler/validate.ts:38]: Basically it's replacing the current task with the new one. But what we need is to transform the previous one into `TaskHistory` _and_ add the new one. I didn't go with that route as it would require some broader changes which would affect the other "flows". While with this change we only need to concern ourselves with the `Approval` flow

**Review by rikukissa — APPROVED** [review]
